weblate/api/views.py

```diff
@@ -1212,6 +1212,45 @@ def get_serializer(self, instance, *args, **kwargs):
     def get_queryset(self):
         return Unit.objects.filter_access(self.request.user).order_by("id")
 
+    @action(detail=True, methods=["post"])
+    def screenshots(self, request, **kwargs):
+        obj = self.get_object()
+
+        if not request.user.has_perm("unit.edit", obj.translation):
+            raise PermissionDenied()
+
+        if "screenshot_id" not in request.data:
+            raise ValidationError({"screenshot_id": "This field is required."})
+
+        try:
+            screenshot = obj.translation.screenshot_set.get(
+                pk=int(request.data["screenshot_id"])
+            )
+        except (Screenshot.DoesNotExist, ValueError) as error:
+            raise ValidationError({"screenshot_id": str(error)})
+
+        obj.units.add(screenshot)
```

Comment:
This looks wrong - `obj` is `Unit` here, it probably shoule be:

```suggestion
        screenshot.units.add(obj)
```